Mathlib/CategoryTheory/ClosedUnderIsomorphisms.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Iso
+import Mathlib.Order.Basic
+
+/-! # Predicates on objects which are closed under isomorphisms
+
+This file introduces the type class `ClosedUnderIsomorphisms P` for predicates
+`P : C → Prop` on the objects of a category `C`.
+
+-/
+
+namespace CategoryTheory
+
+variable {C : Type*} [Category C] (P Q : C → Prop)
+
+/-- A predicate `C → Prop` on the objects of a category is closed under isomorphisms
+if whenever `P X`, then all the objects `Y` that are isomorphic to `X` also satisfy `P Y`. -/
+class ClosedUnderIsomorphisms : Prop where
+  mem_of_iso {X Y : C} (_ : X ≅ Y) (_ : P X) : P Y
```

Comment:
Should `mem_of_iso` be renamed now that we are talking about predicates rather than sets?

Reply:
Perhaps even just `of_iso`?

Reply:
I have renamed `subset_isoClosure` as `le_isoClosure`, etc. The `mem_of_iso` field of `ClosedUnderIsomorphisms` is also renamed `of_iso`. But then, for the API lemmas in the `CategoryTheory` namespace, as it is no longer in a namespace `Set`, I do not see significantly better name to for `CategoryTheory.mem_of_iso` and a few others. (Anyway, if better names are found later, it will be easy to make the change.)